### PR TITLE
[boschshc] Support for Universal Switch I + II

### LINKED ISSUE #16244: [boschshc] Support for Universal Switch II
Hello,

i bought a Universal Switch II.
When i made a Scan it didn't apear and in the openhab logs there was no "Added new thing".

I am using openhab Version 4.1.0 Release and Bosch Smart Home Controller II.

Would appreciate for extension of boschshc binding with new devices from Bosch Smart Home,
Winfried

### PR DESCRIPTION
**Pull request by david-pace**
- add thing type and channel type definitions
- re-generate i18n file
- add constants
- add model classes and enums
- implement service and handlers
- register handlers in factory
- register devices in discovery

closes #16244

### CODE REVIEW

**Review by GerdZanker — APPROVED** [review]
I have no Universal Switches and therefore only did a review of the files.
You changes are as usual: consistent, clean with comments and unit tests  - very good work.
Thank you @david-pace

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/universalswitch/UniversalSwitchHandler.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    private void updateChannels(KeypadServiceState keypadServiceState) {
+        updateState(CHANNEL_KEY_CODE, new DecimalType(keypadServiceState.keyCode));
+        updateState(CHANNEL_KEY_NAME, new StringType(keypadServiceState.keyName.toString()));
+        updateState(CHANNEL_KEY_EVENT_TYPE, new StringType(keypadServiceState.eventType.toString()));
+
+        Instant instant = Instant.ofEpochMilli(keypadServiceState.eventTimestamp);
+        updateState(CHANNEL_KEY_EVENT_TIMESTAMP,
+                new DateTimeType(ZonedDateTime.ofInstant(instant, ZoneId.systemDefault())));
```
Comment (lsiepel) [inline bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/universalswitch/UniversalSwitchHandler.java:63]: Was it intended to use the system default and not the timezone provided by the openHAB `TimeZoneProvider` ?

File: bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -606,4 +642,32 @@
...
+		<item-type>String</item-type>
+		<label>Key Name</label>
+		<description>Name of the key that was pressed.</description>
+		<state readOnly="true"/>
+	</channel-type>
+
+	<channel-type id="key-event-type">
+		<item-type>String</item-type>
+		<label>Key Event Type</label>
+		<description>Indicates how the key was pressed. Possible values are PRESS_SHORT, PRESS_LONG and PRESS_LONG_RELEASED.</description>
```
Comment (lsiepel) [inline bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/thing/thing-types.xml:662]: For your consideration: possible states can be listed as an option:
https://www.openhab.org/docs/developer/bindings/thing-xml.html#state-description

**Review by lsiepel — COMMENTED** [review]
Left a question and minor comment.

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/universalswitch/UniversalSwitchHandler.java (hunk 1 above)
Comment (david-pace) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/universalswitch/UniversalSwitchHandler.java:63]: Thanks for the hint. I was not aware of this provider. I enhanced the code already but need more time to test again with the hardware. Once this is done I will push a new commit.

File: bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/thing/thing-types.xml (hunk 2 above)
Comment (david-pace) [reply bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/thing/thing-types.xml]: Thanks for the hint, I added options where applicable :+1:

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/universalswitch/UniversalSwitchHandler.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/universalswitch/UniversalSwitchHandler.java]: Perfect.

**Review by lsiepel — APPROVED** [review]
LGTM, Let me know when testing is finished and i'll merge the PR.

**Comment by david-pace** [comment]
Hi @lsiepel, I tested the code well with Universal Switch II. It would be good if someone could test with Universal Switch I as well. But of the testers I know, no one has a first generation device.

On the other hand, Universal Switch I has the very same service API, just the button names are different and I used the ones specified in the official [Bosch API](https://local.apidocs.bosch-smarthome.com/#/States/get_devices__deviceId__services_Keypad_state) and also confirmed with a [Python implementation](https://github.com/tschamm/boschshcpy/blob/master/boschshcpy/services_impl.py) (line 625). So I'm fairly confident that this will work, too.

If you agree that the two sources are enough, we can merge this PR 🚀

**Comment by lsiepel** [comment]
> Hi @lsiepel, I tested the code well with Universal Switch II. It would be good if someone could test with Universal Switch I as well. But of the testers I know, no one has a first generation device.
> 
> On the other hand, Universal Switch I has the very same service API, just the button names are different and I used the ones specified in the official [Bosch API](https://local.apidocs.bosch-smarthome.com/#/States/get_devices__deviceId__services_Keypad_state) and also confirmed with a [Python implementation](https://github.com/tschamm/boschshcpy/blob/master/boschshcpy/services_impl.py) (line 625). So I'm fairly confident that this will work, too.
> 
> If you agree that the two sources are enough, we can merge this PR 🚀

As it is a new supported device, there is no risc of regression and we have some milestones before the next stable, hopefully it will get some mileage.

**Pull request merged**